src/sql_types/src/lib.rs

```diff
@@ -96,14 +96,20 @@ pub trait Serializer {
     fn des(&self, out_value: &[u8]) -> String;
 }
 
-struct SmallIntTypeConstraint;
+struct SmallIntTypeConstraint {
+    min: u16,
+}
 
 impl Constraint for SmallIntTypeConstraint {
     fn validate(&self, in_value: &str) -> Result<(), ConstraintError> {
-        match lexical::parse::<i16, _>(in_value) {
-            Ok(_) => Ok(()),
-            Err(e) if e.code == lexical::ErrorCode::InvalidDigit => Err(ConstraintError::NotAnInt),
-            Err(_) => Err(ConstraintError::OutOfRange),
+        if self.min < lexical::parse::<u16, _>(in_value).unwrap() {
```

Comment:
Could you please move `if` check inside `Ok(parsed)` branch, `unwrap`ing will panic and `in_value` will be parsed only once

Reply:
Sure, thanks.